lib/ide-haskell.coffee

```diff
@@ -14,23 +14,29 @@ configDefaults =
 
 _isCabalProject = false         # true if cabal project
 _pluginManager = null           # plugin manager
+_disposables = new CompositeDisposable
 
 activate = (state) ->
-  _isCabalProject = isCabalProject()
+  projRoot = getCabalProjectDir()
+  _isCabalProject = (projRoot != null)
   $(window).on 'focus', updateMenu
   return unless _isCabalProject
 
+  # store project root on the atom project, since the utils need it
+  atom.project.cabalProjectRoot = projRoot
```

Comment:
That's another option.  Just requires another accessor for the singleton.  The nice thing about atom.project is that its already available.  But yeah, storing stuff on the project is ಠ_ಠ, so I'll get rid of it.

BTW, just updated the PR with the autocomplete reg fixes, still working on the WeakMap stuff.